### PR TITLE
Conditional formatting basic

### PR DESCRIPTION
**Pull request by Rick-Kirkham**
There will be another snippet with more advanced conditional formatting APIs dealing with priority of overlapping conditional formatting.

### CODE REVIEW

File: samples/excel/30-range/conditional-formatting-basic.yaml [hunk 1]
```diff
@@ -0,0 +1,390 @@
...
+                criteria[2].formula = "=1000";
+
+                // Cell values between 701 and 1000 get a yellow flag
+                criteria[1].type = Excel.ConditionalFormatIconRuleType.number;
+                criteria[1].operator = Excel.ConditionalIconCriterionOperator.greaterThan;
+                criteria[1].formula = "=700";
+
+                /*
+                    Unless you are doing something special with the "low" icon criterion
+                    (criteria[0]) , you  do not need to explicitly set it's properties. Any cells
```
Comment (Zlatkovsky) [inline samples/excel/30-range/conditional-formatting-basic.yaml]: nit: space after the `criteria[0])` before the comma

File: samples/excel/30-range/conditional-formatting-basic.yaml [hunk 2]
```diff
@@ -0,0 +1,390 @@
...
+                    that the middle and top criteria do not apply to will get the "low" icon. In
+                    this snppet everything below 700 gets a red flag.
```
Comment (Zlatkovsky) [inline samples/excel/30-range/conditional-formatting-basic.yaml]: spelling: `snppet`

File: samples/excel/30-range/conditional-formatting-basic.yaml [hunk 3]
```diff
@@ -0,0 +1,390 @@
...
+        }
+
+        async function applyTextFormat() {
+            await Excel.run(async (context) => {
+                const sheet = context.workbook.worksheets.getItem("Sample");
+                const range = sheet.getRange("B16:D18");
+                const conditionalFormat = range.conditionalFormats
+                    .add(Excel.ConditionalFormatType.containsText);
+                conditionalFormat.textComparison.format.font.color = "red";
+                conditionalFormat.textComparison.rule = {operator: Excel.ConditionalTextOperator.contains, text: "Delayed"};
```
Comment (Zlatkovsky) [inline samples/excel/30-range/conditional-formatting-basic.yaml]: I would split this over 2 lines, and also have space before the closing brace.  JS style is

    a = { b: 5 }

File: samples/excel/30-range/conditional-formatting-basic.yaml [hunk 4]
```diff
@@ -0,0 +1,390 @@
...
+                $("#conditional-formats li").remove();
+                if (worksheetRange.conditionalFormats.items.length > 0) {
+                    for (let i = 0; i < worksheetRange.conditionalFormats.items.length; i++) {
+                        let $a = $(`<a href=''>${worksheetRange.conditionalFormats.items[i].type}</a>`)
+                            .click(
+                                async (e: any) => {
+                                    e.preventDefault();
+                                    const sheet = context.workbook.worksheets
+                                        .getItem(e.data.worksheet);
+                                    const conditionalFormat = sheet.getRange()
```
Comment (Zlatkovsky) [inline samples/excel/30-range/conditional-formatting-basic.yaml]: Not in love with getItemAt if we can avoid it.  Can we?

File: samples/excel/30-range/conditional-formatting-basic.yaml [hunk 5]
```diff
@@ -0,0 +1,390 @@
...
+
+                $(".conditional-formats").hide();
+            });
+        }
+
+        async function setup() {
+            await Excel.run(async (context) => {
+                const sheet = await OfficeHelpers.ExcelUtilities
+                    .forceCreateSheet(context.workbook, "Sample");
+                queueCommandsToCreateTemperatureTable(sheet);
```
Comment (Zlatkovsky) [inline samples/excel/30-range/conditional-formatting-basic.yaml]: I'm fine with dropping the `queueCommandsTo` and just doing `createTemperatureTable`

File: samples/excel/30-range/conditional-formatting-basic.yaml (hunk 4 above)
Comment (Rick-Kirkham) [reply samples/excel/30-range/conditional-formatting-basic.yaml]: I don't see an obvious way to not use it. In fact, this looks like a scenario that getItemAt was invented for. ConditionalFormats don't have a name property or any other intrinsically unique property. BTW, what's wrong with getItemAt?

File: samples/excel/30-range/conditional-formatting-basic.yaml (hunk 5 above)
Comment (Rick-Kirkham) [reply samples/excel/30-range/conditional-formatting-basic.yaml]: I've grown fond of it. :-)

**Comment by Rick-Kirkham** [comment]
Regarding @jakobpn 's snippet. I didn't notice till now that he'd submitted a PR for it. I was sent it as a gist to use as a starting point. The gist ID didn't tell me who wrote it. I used Jakob's list-all and clear-all methods.

**Comment by Zlatkovsky** [comment]
@Rick-Kirkham , let's just discuss this in person and then we can merge in.
I'm likely WFH tomorrow (Tuesday), but will be at work later in the week.  Feel free to just set up 15 minutes.

**Pull request merged**